src/structures/timedcache.ts

```diff
@@ -0,0 +1,103 @@
+interface ITimedValue<V> {
+    value: V;
+    ts: number;
+}
+
+export class TimedCache<K, V> implements Map<K, V> {
+    private readonly  map: Map<K, ITimedValue<V>>;
+
+    public constructor(private readonly liveFor: number) {
+        this.map = new Map();
+    }
+
+    public clear(): void {
+        this.map.clear();
+    }
+
+    public delete(key: K): boolean {
+        return this.map.delete(key);
+    }
+
+    public forEach(callbackfn: (value: V, key: K, map: Map<K, V>) => void): void {
+        for (const item of this) {
+            callbackfn(item[1], item[0], this);
+        }
+    }
+
+    public get(key: K): V | undefined {
+        const v = this.map.get(key);
+        if (v) {
```

Comment:
everywhere else we check for `!== undefined`, why not here? This would cause things to break in case of `null` or `false`

Reply:
Good catch!